### ISSUE TITLE
Proactively check trace and span lengths

### ISSUE THREAD

**Issue by agardnerIT** [agardnerIT/tracepusher#29]
Almost identical to https://github.com/agardnerIT/logpusher/issues/2 - tracepusher should warn users and either proactively fail (do not even attempt the collector push OR at least warn that users SHOULD expect a failure). The implementation is left up to whomever picks up this issue.

Summary: OTEL expects certain trace and span lengths. The collector will fail to accept the trace if they aren't the correct length. So tracepusher should be "length-aware" for user inputs and warn / fail.

Documentation should be updated to mention this "rule".

**Comment by agardnerIT** [comment]
@bradmccoydev - I know you've hit this already.

**Comment by agardnerIT** [comment]
If anyone would like to take this up, the improvement is in this area: https://github.com/agardnerIT/tracepusher/blob/main/tracepusher.py#L137-L140

`len(trace_id) ==32` and `len(span_id)==16` otherwise they're invalid.

**Comment by bradmccoydev** [comment]
@MilijanBogdanovic is keen to help on this issue, can you please assign it to him

**Comment by bogdanovicdev** [comment]
Yes, looking forward to working on this, thanks

**Comment by agardnerIT** [comment]
Hi @MilijanBogdanovic, just checking in to see if you need anything from me on this? It'd be good to get this into the v0.8.0 release & self-contained binaries (assuming the [release candidate](https://github.com/agardnerIT/tracepusher/releases/tag/0.8.0-rc1) [feedback ](https://github.com/agardnerIT/tracepusher/issues/46)is good for the binaries and they're ready to go).

**Comment by bogdanovicdev** [comment]
Hey @agardnerIT 
I have added a PR for this issue, my question is just since I'm new to contributing open source, what is your testing criteria before merging? How do you usually test features on this repo before merging?
Thanks, sorry for wait,
looking forward to helping you further

**Comment by agardnerIT** [comment]
Thanks! I've tested this manually in a codespace and merged it. I admit though, testing is entirely lacking and somewhere I'd like assistance (if you feel comfortable in that area).

For my testing I just ran tracepusher in dry run and debug mode:

```
python tracepusher.py -ep https://dummy -spn span1 -sen serviceA -dur 2 --debug true --dry-run true --trace-id abc123 --span-id abc123
```

Fed it some obviously incorrect trace and span ID lengths. Then followed [the documented openssl way to generate correct lengths](https://agardnerit.github.io/tracepusher/reference/multi-span-traces/) and checked they passed:

```
trace_id=$(openssl rand -hex 16)
span_id=$(openssl rand -hex 8)
python tracepusher.py -ep https://dummy -spn span1 -sen serviceA -dur 2 --debug true --dry-run true --trace-id $trace_id--span-id $span_id
```

I'd loe to automate some of this testing though if you know how!

Closing this as done via #47
